app/src/main/java/com/sun/ise/ui/detail/DetailActivity.kt

```diff
@@ -0,0 +1,47 @@
+package com.sun.ise.ui.detail
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import com.google.android.material.appbar.AppBarLayout
+import com.google.android.material.tabs.TabLayout
+import com.sun.ise.R
+import com.sun.ise.ui.common.ViewPagerAdapter
+import com.sun.ise.ui.detail.description.DescriptionFragment
+import com.sun.ise.ui.detail.feature.FeatureFragment
+import com.sun.ise.ui.detail.requirement.RequirementFragment
+import kotlinx.android.synthetic.main.activity_detail.*
+
+class DetailActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_detail)
+        setupAppBar()
+        setupTabLayout()
+    }
+
+    private fun setupAppBar() {
+        appBarLayout.addOnOffsetChangedListener(object : AppBarLayout.OnOffsetChangedListener {
+            var isShow = false
+            var scrollRange = -1
+
+            override fun onOffsetChanged(appBarLayout: AppBarLayout?, verticalOffset: Int) {
+                if (scrollRange == -1)
```

Comment:
đoạn code này ntn ấy em nhỉ
e vừa khai báo scrollRange - -1 ở trên.
ở dưới lại check lại scrollragne làm gì

Reply:
cho a hỏi ý nghĩa của đoạn code này nữa nhé

Reply:
Ban đầu em chưa hiểu rõ về trạng thái của appBarLayout nên đặt thừa 2 biến isShow với scrollRange ạ.

Với method onOffsetChanged thì nó sẽ xác định trạng thái của appBarLayout bằng param verticalOffset. Nếu verticalOffset == 0 thì appbar sẽ mở toàn bộ và ngược lại.